### PR TITLE
Features/sma 329 improve repo

### PR DESCRIPTION
**Pull request by Aboudjem**
Add Audits folder

Improved Readme

Add contributions / Readme/ Licenses

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,86 @@
-# biconomy-paymasters
+# Biconomy Paymasters 🌐
 
-### Playground for ERC4337 Paymasters built with :heart_eyes: by Biconomy 
-#### Some examples 
+Welcome to the `biconomy-paymasters` repository! Paymaster contracts enable seamless transaction fee handling in decentralized applications. 🛠️
 
-1. Singleton Verifying Paymaster: Acts as a sponsorship paymaster and lets Dapps manage deposit without deploying a new one for each Dapp.
+## What is Biconomy Paymasters? 🤔
```
Comment (livingrockrises) [inline README.md:5]: is -> are

would be good to talk about ERC4337 paymasters on the high level maybe with some flow diagrams.

File: README.md [hunk 2]
```diff
@@ -1,87 +1,86 @@
...
 
-2. Token Paymaster: Extended version of Verifying Paymaster which can accept fees from user by withdrawing ERC20 tokens
+Biconomy Paymasters are smart contracts that abstract away the complexity of gas fees for end-users. By utilizing these contracts, developers can offer their users gasless transactions or the ability to pay for gas in ERC20 tokens. 🚀
```
Comment (livingrockrises) [inline README.md:7]: can just limit to gas abstraction. then talk about kind of paymaster we already have or we plan to have with gas abstraction (how) aspect of each

File: README.md [hunk 3]
```diff
@@ -1,87 +1,86 @@
...
+- ERC 4337 Account Abstraction compliant.
+- Multi-token support for gas payments.
 
-Custom versions of above Verifying (allow Dapp deposit sponsorship in different tokens) and Token Paymaster (acts as deposit paymaster also)
+### ERC20 Token Paymaster
 
+![ERC20 Token Paymaster](./assets/readme/erc20-token-gas-payment-flow.png)
 
+- ERC20 Token Paymaster helps users pay for their transactions using ERC20 tokens.
+- Users initiate a transaction using an ERC20 token.
```
Comment (livingrockrises) [inline README.md:23]: refer to ERC20 paymaster design notion doc

File: README.md (hunk 1 above)
Comment (livingrockrises) [reply README.md:5]: check some content from here

https://erc4337.mirror.xyz/7DUTUn2eNrjvum3tWAnRih8576IrX13E6WnZeADvAHQ

https://erc4337.mirror.xyz/RKG9kt7af3B_Dj0KHjuhwOOpEP6JxLXr-2vW4DOlhQM

there are also some articles by alchemy and visa

**Review by livingrockrises — COMMENTED** [review]
reviewed

**Comment by livingrockrises** [comment]
in one of the diagrams it's not clear what SmartAccount is and how it would return signedUserOp

**Comment by livingrockrises** [comment]
resolve conflicts

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Contributing to Biconomy Projects 🚀
+
+First off, thank you for considering contributing to Biconomy! We truly appreciate your effort and contributions from the community are what make Biconomy awesome. 🙌
```
Comment (filmakarov) [inline CONTRIBUTING.md:3]: I suggest dividing this into two:

 We truly appreciate your effort. Contributions from the community are what make Biconomy awesome 🙌

File: README.md [hunk 5]
```diff
@@ -1,87 +1,86 @@
...
 
-1. Singleton Verifying Paymaster: Acts as a sponsorship paymaster and lets Dapps manage deposit without deploying a new one for each Dapp.
+## What is Biconomy Paymasters? 🤔
 
-2. Token Paymaster: Extended version of Verifying Paymaster which can accept fees from user by withdrawing ERC20 tokens
+Biconomy Paymasters are smart contracts that abstract away the complexity of gas fees for end-users. By utilizing these contracts, developers can offer their users gasless transactions or the ability to pay for gas in ERC20 tokens. 🚀
 
-#### Other WIP
+- **Sponsorship Paymaster**: Allows transactions without end-users needing to pay for gas, enhancing UX.
+- **Token Paymaster**: Provides the ability to pay for transactions with ERC20 tokens.
```
Comment (filmakarov) [inline README.md:10]: Maybe highlight, that users still pay by themselves with token paymaster and token paymaster allows not having native token

File: README.md [hunk 6]
```diff
@@ -1,87 +1,86 @@
...
+- Multi-token support for gas payments.
 
-Custom versions of above Verifying (allow Dapp deposit sponsorship in different tokens) and Token Paymaster (acts as deposit paymaster also)
+### ERC20 Token Paymaster
 
+![ERC20 Token Paymaster](./assets/readme/erc20-token-gas-payment-flow.png)
 
+- ERC20 Token Paymaster helps users pay for their transactions using ERC20 tokens.
+- Users initiate a transaction using an ERC20 token.
+- Paymaster validates the transaction and forwards it to the network while handling necessary fee conversions.
```
Comment (filmakarov) [inline README.md:24]: Technically Paymaster doesn't forward the transaction. It just pays for the userOp.

File: README.md [hunk 7]
```diff
@@ -1,87 +1,86 @@
...
 
-# How to run the project
+### Sponsorship Paymaster
 
-#### You're going to need to place a private key in a .env file in the root. ####
+![Sponsorship Paymaster](./assets/readme/paymaster-off-chain-request-processing-flow.png)
 
-#### In order to add/udpate git submodule account-abstraction: ####
-.gitmodules file is already added. two submodules are being used in this project
+- Sponsorship Paymaster covers transaction fees for users.
```
Comment (filmakarov) [inline README.md:31]: I suggest changing txns to userOps everywhere

File: README.md [hunk 8]
```diff
@@ -1,87 +1,86 @@
...
+- The process starts with the user initiating a transaction.
+- Paymaster takes over and sponsors the fees, so users don't have to bear the gas costs.
```
Comment (filmakarov) [inline README.md:33]: Should we describe that to sponsor a userOp, paymaster still needs signature by verifying party that reviews userOps before agreeing to pay for it

File: README.md (hunk 7 above)
Comment (filmakarov) [reply README.md:31]: Maybe I'm missing something, but I find it a bit confusing, that we have it named 'Sponsorship Paymaster', however I can't find Sponsorship Paymaster in the repo.
I understand Verifying Paymaster is not the best name, Sponsorship Paymaster is much better , but I think need some consistency between readme and file names here

**Review by filmakarov — APPROVED** [review]
Approving to allow for the renewed Readme to be displayed for users.
However please consider iterating on my suggestions

File: README.md (hunk 7 above)
Comment (Aboudjem) [reply README.md:31]: You're right @filmakarov, I found the references to 'Sponsorship Paymaster' in both Biconomy [blog](https://www.biconomy.io/post/how-biconomy-paymaster-works-sponsor-gas-or-pay-in-any-token) and [official documentation](https://docs.biconomy.io/Paymaster/description#sponsorship-paymaster). 

I agree 'Sponsorship Paymaster' sounds like a better name to me too. Let's bring this up with the team for further discussion.

**Review by livingrockrises — APPROVED** [review]

**Pull request merged**